Client.java

```diff
@@ -0,0 +1,170 @@
+    BuildJson buildJson = new BuildJson();
+    Request request = new Request();
+
+    Scanner scan = new Scanner(System.in);
+    String[] split = System.console().readLine().split(" ");
+    System.out.println();
+
+    if(split[COMMAND_LOC].toLowerCase().equals(START)) {
+      if(split.length < MIN_ARGS) {
+        throw new Exception("No process provided");
+      }
+      String completeJob = split[PID_OR_PROCESS_LOC];
+      for(int i = PID_OR_PROCESS_LOC+1; i < split.length; i++) {
+        completeJob += (" " + split[i]);
+      }
+      request.setProcess(completeJob);
+      json = buildJson.buildRequest(request, START);
+    } else if(split[COMMAND_LOC].toLowerCase().equals(STOP)) {
+      if(split.length < MIN_ARGS) {
+        throw new Exception("No PID provided");
+      }
+      long pid = Long.parseLong(split[PID_OR_PROCESS_LOC]);
+      request.setPid(pid);
+      json = buildJson.buildRequest(request, STOP);
+    } else if(split[COMMAND_LOC].toLowerCase().equals(QUERY)) {
+      if(split.length < MIN_ARGS) {
+        throw new Exception("No PID provided");
+      }
+      long pid = Long.parseLong(split[PID_OR_PROCESS_LOC]);
+      request.setPid(pid);
+      json = buildJson.buildRequest(request, QUERY);
+    } else if(split[COMMAND_LOC].toLowerCase().equals(GET_CURRENT)) {
+      json = buildJson.buildRequest(request, GET_CURRENT);
+    } else if(split[COMMAND_LOC].toLowerCase().equals(OUTPUT)) {
+      if(split.length < MIN_ARGS) {
+        throw new Exception("No PID provided");
+      }
+      long pid = Long.parseLong(split[PID_OR_PROCESS_LOC]);
+      request.setPid(pid);
+      json = buildJson.buildRequest(request, OUTPUT);
+    } else {
+      throw new Exception("No valid command provided");
+    }
+    return json;
+  }
+
+  private static ParsedResponse sendJsonRequest(JSONObject jsonRequest) throws Exception {
+    Credentials credentials = new Credentials();
+    SSLContext sslContext = credentials.init(CLIENT_KEY_LOC);
+
+    SSLSocketFactory sslSocketFactory = sslContext.getSocketFactory();
+    SSLSocket sslSocket = (SSLSocket)sslSocketFactory.createSocket(HOST, PORT);
+
+    sslSocket.setEnabledCipherSuites(sslSocket.getSupportedCipherSuites());
+    sslSocket.startHandshake();
+
+    SSLSession sslSession = sslSocket.getSession();
+
+    OutputStream outputStream = sslSocket.getOutputStream();
+    ObjectOutputStream objectOutputStream = new ObjectOutputStream(outputStream);
+    objectOutputStream.writeObject(jsonRequest.toString());
+    outputStream.flush();
+
+    InputStream inputStream = sslSocket.getInputStream();
+    ObjectInputStream objectInputStream = new ObjectInputStream(inputStream);      
+   
+    JSONObject jsonObjParsed = new JSONObject((String)objectInputStream.readObject());
+    ParseJson parseJson = new ParseJson();
+    HashMap<String, Object> responseMap = parseJson.parseJson(jsonObjParsed);
+
+    return new ParsedResponse(responseMap, sslSocket);
```

Comment:
I just realized that socket will not be needed outside this method - do you think you need to have it around and have to explicitly close it only after handling the response?
This equally applies to `ParsedRequest`.
Sorry if my previous comment was confusing - I sketched the code in a hurry.

Reply:
I have fixed this issued in ParsedResponse. The socket is needed in multiple methods in the server so I have not made the equivalent change to ParsedRequest.